### PR TITLE
feat: add support for start day of week on MLT axis

### LINKED ISSUE #2361: Multilayer time axis should read `dow` from moment locale
**Is your feature request related to a problem? Please describe.**

In the multilayer time axis we hard code the start day of the week to `1` (aka Monday). This does not allow the chart consumer to change this to adapt to the data.

https://github.com/elastic/elastic-charts/blob/3bddec8dd263f516db6aa2ff45830583464d008e/packages/charts/src/chart_types/xy_chart/axes/timeslip/continuous_time_rasters.ts#L320

#### Demo

Zight Recording 2024-03-14 at 02 31 47 PM

See demo [here](https://codesandbox.io/p/sandbox/dow-demo-x66ff7?file=%2Fsrc%2FApp.tsx%3A19%2C35)

**Describe the solution you'd like**
The start day of the week should be driven by the [`moment.Locale#week.dow`](https://momentjs.com/docs/#/customization/dow-doy/)

**Describe alternatives you've considered**
N/A

**Additional context**
This only applies to the tick placement and labels, not the placement of the rendered geometries.

**Kibana Cross Issues**

https://github.com/elastic/kibana/issues/178115

**Checklist**

- [x] this request is checked against already exist requests
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

This PR adds support to set the start day of the week for multilayer time axis via `Settings.dow`. See [story](https://ech-e2e-ci--pr-2362-8wz1o55c.web.app/?path=/story/test-cases--start-day-of-week&globals=toggles.showHeader:true;toggles.showChartTitle:false;toggles.showChartDescription:false;toggles.showChartBoundary:false;theme:light) example for demo below ⬇️ 

Zight Recording 2024-03-18 at 02 34 49 PM

## Details

The `Settings.dow` option expects an integer between `1` and `7` according to [**ISO 8601**](https://en.wikipedia.org/wiki/ISO_week_date)

## Issues

closes #2361

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Comment by nickofthyme** [comment]
@markov00 I updated the code based on our discussion to move the option to the `Settings`.

File: e2e/screenshots/all.test.ts-snapshots/baselines/test-cases/start-day-of-week-chrome-linux.png
Comment (markov00) [inline e2e/screenshots/all.test.ts-snapshots/baselines/test-cases/start-day-of-week-chrome-linux.png:1]: could you add a test with a sunday start of the week? so we catch both in tests

File: storybook/stories/test_cases/11_start_day_of_week.story.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  return (
+    <>
+      <Chart title={title} description={description}>
+        <Settings dow={startDow} />
+        <Axis id="y" title="Count" position={Position.Left} />
+        <Axis
+          id="x"
+          title="Time"
+          position={Position.Bottom}
+          timeAxisLayerCount={3}
```
Comment (markov00) [inline storybook/stories/test_cases/11_start_day_of_week.story.tsx:66]: please reduce it to 2, will be the default and probably we are not going to use 3 layers in the future

Comment (nickofthyme) [reply storybook/stories/test_cases/11_start_day_of_week.story.tsx]: Done in f50fcab

File: e2e/screenshots/all.test.ts-snapshots/baselines/test-cases/start-day-of-week-chrome-linux.png
Comment (nickofthyme) [reply e2e/screenshots/all.test.ts-snapshots/baselines/test-cases/start-day-of-week-chrome-linux.png:1]: Done in f50fcab

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Review by markov00 — APPROVED** [review]

**Pull request merged**